pkg/agent/nodeportlocal/k8s/handlers.go

```diff
@@ -32,123 +32,89 @@ func (c *Controller) addRuleForPod(pod *corev1.Pod) error {
 
 	for _, container := range podContainers {
 		for _, cport := range container.Ports {
-			if c.portTable.RuleExists(podIP, int(cport.ContainerPort)) {
+			port := int(cport.ContainerPort)
+			if c.portTable.RuleExists(podIP, port) {
 				continue
 			}
-			port := fmt.Sprint(cport.ContainerPort)
-			nodePort, err := c.portTable.AddRule(podIP, int(cport.ContainerPort))
+			nodePort, err := c.portTable.AddRule(podIP, port)
 			if err != nil {
 				return err
 			}
-			assignPodAnnotation(pod, port, nodeIP, fmt.Sprint(nodePort))
+			assignPodAnnotation(pod, nodeIP, port, nodePort)
 		}
 	}
 	return nil
 }
 
-// HandleAddPod handles Pod annotations in NPL for an added pod.
-func (c *Controller) HandleAddPod(key string, obj interface{}) error {
-	oldObj, exists, _ := c.OldObjStore.GetByKey(key)
-	if exists {
-		defer c.OldObjStore.Delete(oldObj)
-		return c.HandleUpdatePod(oldObj, obj)
-	}
-
-	pod := obj.(*corev1.Pod).DeepCopy()
-	klog.Infof("Got add event for pod: %s/%s", pod.Namespace, pod.Name)
-	err := c.addRuleForPod(pod)
-	if err != nil {
-		return err
-	}
-	if pod.Annotations[NPLAnnotationStr] != "" {
-		return c.updatePodAnnotation(pod)
-	}
-	return nil
-}
-
 // HandleDeletePod handles pod annotations for a deleted pod.
 func (c *Controller) HandleDeletePod(key string) error {
-	obj, exists, err := c.OldObjStore.GetByKey(key)
-	if err != nil {
-		return err
-	}
-	if !exists {
-		klog.Infof("Could not find old object for key: %v", key)
-		return nil
-	}
-	defer c.OldObjStore.Delete(obj)
-	pod := obj.(*corev1.Pod)
-	klog.Infof("Got delete event for pod: %s/%s", pod.Namespace, pod.Name)
-	podIP := pod.Status.PodIP
-	if podIP == "" {
-		klog.Infof("IP address not found for pod: %s/%s", pod.Namespace, pod.Name)
+	klog.Infof("Got delete event for pod: %s", key)
```

Comment:
Do you want to generate an INFO log message for every Pod add/delete event? If not, consider adding a verbosity level, like klog.V(2).Infof().

Reply:
Would it be fine for delete events to have logs ? @antoninbas had suggested this.